### PR TITLE
Feat/hypothetical message

### PR DESCRIPTION
**Pull request by ruthenian8**
# Description

Add multimessage class for compatibility with  DeepPavlov dream.

# Checklist

- [x] I have covered the code with tests
- [x] I have added comments to my code to help others understand it
- [x] I have updated the documentation to reflect the changes
- [x] I have performed a self-review of the changes

### CODE REVIEW

File: dff/script/core/message.py [hunk 1]
```diff
@@ -103,3 +103,7 @@ class Message(DataModel):
 
     def __repr__(self) -> str:
         return " ".join([f"{key}='{value}'" for key, value in self.dict(exclude_none=True).items()])
+
+
+class MultiMessage(Message):
+    hypotheses: Optional[list] = None
```
Comment (kudep) [inline dff/script/core/message.py:109]: Optional[list[Message]]

**Review by kudep — CHANGES_REQUESTED** [review]

File: dff/script/core/message.py (hunk 1 above)
Comment (ruthenian8) [reply dff/script/core/message.py]: Changed

File: dff/script/core/message.py [hunk 2]
```diff
@@ -103,3 +103,7 @@ class Message(DataModel):
 
     def __repr__(self) -> str:
         return " ".join([f"{key}='{value}'" for key, value in self.dict(exclude_none=True).items()])
+
+
+class MultiMessage(Message):
+    hypotheses: Optional[List[Message]] = None
```
Comment (RLKRo) [inline dff/script/core/message.py:109]: Maybe make `confidences` a field of `MultiMessage`?
```suggestion
    hypotheses: Optional[List[Message]] = None
    confidences: Optional[List[Message]] = None
```

File: examples/script/responses/4_multi_message.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    ),  # node1 -> node2
+    (
+        Message(text="Let's talk about music."),
+        Message(text="Sorry, I can not talk about music now."),
+    ),  # node2 -> node3
+    (Message(text="Ok, goodbye."), Message(text="BYE")),  # node3 -> node4
+)
+
+# %%
+random.seed(31415)  # predestination of choice
```
Comment (RLKRo) [inline examples/script/responses/4_multi_message.py:200]: Is it used here?

File: examples/script/responses/4_multi_message.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            # If "Hi" == request of user then we make the transition
+        },
+        "node1": {
+            RESPONSE: MultiMessage(
+                messages=[
+                    Message(text="Hi, what is up?", misc={"confidences": 0.85}),
+                    Message(text="Hello, how are you?", misc={"confidences": 0.9}),
+                ]
+            ),
+            # Random choice from candicate list.
```
Comment (RLKRo) [inline examples/script/responses/4_multi_message.py:93]: Does it do a random choice?
```suggestion
```

File: examples/script/responses/4_multi_message.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+from dff.utils.testing.common import (
+    check_happy_path,
+    is_interactive_mode,
+    run_interactive_mode,
+)
+
+
+# %% [markdown]
+"""
+The response can be set by any object of python:
```
Comment (RLKRo) [inline examples/script/responses/4_multi_message.py:27]: By Callable or Message.

File: examples/script/responses/4_multi_message.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+* Callable objects. If the object is callable it must have a special signature:
+
+        func(ctx: Context, actor: Actor, *args, **kwargs) -> Any
+
+* Non-callable objects. If the object is not callable,
+    it will be returned by the agent as a response.
+
+Out of the box `DSL` has a single response function `choice`
+that gives one random response from the list of responses.
```
Comment (RLKRo) [inline examples/script/responses/4_multi_message.py:37]: Remove this, `choice` is not used.
```suggestion
```

**Review by kudep — APPROVED** [review]

**Review by RLKRo — APPROVED** [review]

**Pull request merged**